lib/archethic.ex

```diff
@@ -321,4 +335,132 @@ defmodule Archethic do
     TransactionChain.fetch_size_remotely(address, nodes)
   end
+
+  @doc """
+  Fetch a summaries aggregate for a given date.
+  Check locally first and fallback to a quorum read
+  """
+  @spec fetch_summaries_aggregate(DateTime.t()) ::
+          {:ok, BeaconChain.SummaryAggregate.t()} | {:error, atom()}
+  def fetch_summaries_aggregate(date) do
+    case BeaconChain.get_summaries_aggregate(date) do
+      {:error, :not_exists} ->
+        nodes = P2P.authorized_and_available_nodes()
+        BeaconChain.fetch_summaries_aggregate(date, nodes)
+
+      {:ok, aggregate} ->
+        {:ok, aggregate}
+    end
+  end
+
+  @doc """
+  Request from the beacon chains all the summaries for the given dates and aggregate them
+  """
+  @spec fetch_and_aggregate_summaries(DateTime.t()) :: BeaconChain.SummaryAggregate.t()
+  def fetch_and_aggregate_summaries(date) do
+    BeaconChain.fetch_and_aggregate_summaries(date, P2P.authorized_and_available_nodes())
+  end
+
+  @doc """
+  Retrieve the genesis address locally or remotely
+  """
+  def fetch_genesis_address_remotely(address) do
+    case TransactionChain.get_genesis_address(address) do
+      ^address ->
+        # if returned address is same as given, it means the DB does not contain the value
+        TransactionChain.fetch_genesis_address_remotely(address)
+
+      genesis_address ->
+        genesis_address
+    end
+  end
+
+  @doc """
+  Slots which are already has been added
+  Real time transaction can be get from pubsub
+  """
+  @spec list_transactions_summaries_from_current_slot(DateTime.t()) ::
+          list(TransactionSummary.t())
+  def list_transactions_summaries_from_current_slot(date = %DateTime{} \\ DateTime.utc_now()) do
+    authorized_nodes = P2P.authorized_and_available_nodes()
+
+    ref_time = DateTime.truncate(date, :millisecond)
+
+    next_summary_date = BeaconChain.next_summary_date(ref_time)
+
+    BeaconChain.list_subsets()
+    |> Flow.from_enumerable(stages: 256)
+    |> Flow.flat_map(fn subset ->
+      # Foreach subset and date we compute concurrently the node election
+      subset
+      |> Election.beacon_storage_nodes(next_summary_date, authorized_nodes)
+      |> Enum.filter(&Node.locally_available?/1)
+      |> P2P.nearest_nodes()
+      |> Enum.take(3)
+      |> Enum.map(&{&1, subset})
+    end)
+    # We partition by node
+    |> Flow.partition(key: {:elem, 0})
+    |> Flow.reduce(fn -> %{} end, fn {node, subset}, acc ->
+      # We aggregate the subsets for a given node
+      Map.update(acc, node, [subset], &[subset | &1])
+    end)
+    |> Flow.flat_map(fn {node, subsets} ->
+      # For this node we fetch the summaries
+      fetch_summaries(node, subsets)
+    end)
+    |> Stream.uniq_by(& &1.address)
+    |> Enum.sort_by(& &1.timestamp, {:desc, DateTime})
+  end
+
+  @doc """
+  Check if a transaction exists at address
+  """
+  @spec transaction_exists?(binary()) :: boolean()
+  def transaction_exists?(address) do
```

Comment:
This function can check the DB before asking network, using `TransactionChain.transaction_exists?` if it return false, then we ask the network. This function can check both in db storage or io storage